clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4503,6 +4506,47 @@ void handleFailure(Throwable throwable) {
         return new UpdateFeaturesResult(new HashMap<>(updateFutures));
     }
 
+    @Override
+    public DecommissionBrokerResult decommissionBroker(int brokerId, DecommissionBrokerOptions options) {
+        final KafkaFutureImpl<Void> future = new KafkaFutureImpl<>();
+        final long now = time.milliseconds();
+        final Call call = new Call("decommissionBroker", calcDeadlineMs(now, options.timeoutMs()),
+            new ControllerNodeProvider()) {
```

Comment:
Shouldn't we use `LeastLoadedNodeProvider`? The kip-500 controller will not be reachable anyway.

Reply:
Ah, didn't know a `LeastLoadedNodeProvider` existed. Fixed!